### PR TITLE
aig adding custom cache key

### PR DESCRIPTION
**Pull request by kathayl**
AIG - adding new header for custom cache key

### CODE REVIEW

File: content/ai-gateway/configuration/caching.md [hunk 1]
```diff
@@ -110,3 +110,32 @@ curl https://gateway.ai.cloudflare.com/v1/{account_id}/{gateway_id}/openai/chat/
    	 }
 '
 ```
+
+### Custom cache key (cf-aig-cache-key)
+
```
Comment (mchenco) [inline content/ai-gateway/configuration/caching.md:115]: can we get more info on why someone would want to set a custom cache key / why it matters?

File: content/ai-gateway/configuration/caching.md [hunk 2]
```diff
@@ -110,3 +110,32 @@ curl https://gateway.ai.cloudflare.com/v1/{account_id}/{gateway_id}/openai/chat/
...
+In order to override the default cache key, you can use the header **cf-aig-cache-key**. 
+
+The first time you use the header, you will get a response from the provider. Then, each subsequent time the header is used, you will receive the cached response. If no **cf-cache-ttl header** is used, responses will be cached for 5 minutes. If the **cf-cache-ttl header** is used, responses will be cached according to Cache TTL.
```
Comment (kodster28) [inline content/ai-gateway/configuration/caching.md]: Is `if no **cf-cache-ttl** header is used` part accurate? So setting the cache key overrides your default caching settings regardless?

File: content/ai-gateway/configuration/caching.md [hunk 3]
```diff
@@ -110,3 +110,32 @@ curl https://gateway.ai.cloudflare.com/v1/{account_id}/{gateway_id}/openai/chat/
...
+
+As an example, when submitting a request to OpenAI, include the header in the following manner:
+
+```bash
+---
+header: Request with custom cache key
+---
+
+curl https://gateway.ai.cloudflare.com/v1/{account_id}/{gateway_id}/openai/chat/completions \
+  --header 'Authorization: Bearer $TOKEN' \
```
Comment (kodster28) [inline content/ai-gateway/configuration/caching.md]:
```suggestion
  --header 'Authorization: Bearer {openai_token}' \
```

**Review by kodster28 — APPROVED** [review]

File: content/ai-gateway/configuration/caching.md (hunk 1 above)
Comment (kodster28) [reply content/ai-gateway/configuration/caching.md:115]: From another area of our docs, `custom cache keys let you precisely set the cacheability setting for any resource`

Comment (kodster28) [reply content/ai-gateway/configuration/caching.md:115]: Realize we actually need to add that info to another place as well 🙃 .

#15212

**Comment by kodster28** [comment]
hyperlint review

File: content/ai-gateway/configuration/caching.md (hunk 2 above)
Comment (kathayl) [reply content/ai-gateway/configuration/caching.md]: what i learned the order of operations is:
-if cache-ttl is set, that takes priority
-if it's not, then it's whatever the cache setting the user set in the dashboard is
-if the user had not turned on caching in the dashboard, then it's 5 minutes

How's this rewrite?

The first time you use the header, you will get a response from the provider. Then, each subsequent time the header is used, you will receive the cached response. If the cf-cache-ttl header is used, responses will be cached according to Cache TTL. However, if no cf-cache-ttl header is used, responses will be cached according to dashboard settings. If caching is not turned on for the gateway, responses will be cached for 5 minutes by default.

**Pull request merged**